src/vs/workbench/contrib/accessibility/browser/accessibleView.ts

```diff
@@ -62,13 +63,18 @@ export interface IAccessibleContentProvider {
 	 * When the language is markdown, this is provided by default.
 	 */
 	getSymbols?(): IAccessibleViewSymbol[];
+	/**
+	 * Note that this will only take effect if the provider has an ID.
+	 */
+	onDidRequestClearLastProvider?: Event<AccessibleViewProviderId>;
 }
 
 export const IAccessibleViewService = createDecorator<IAccessibleViewService>('accessibleViewService');
 
 export interface IAccessibleViewService {
 	readonly _serviceBrand: undefined;
 	show(provider: IAccessibleContentProvider): void;
+	showLastProvider(id: AccessibleViewProviderId): void;
```

Comment:
Instead of firing an event with an id back to another provider (circular dep), can we indicate in the `show` call that we want to go back upon exit/close. Then manage it all inside the a11y view service?

Reply:
No, because at the time of the `show` call, we don't know that we want to go back. 

`Show` is called before run recent command is invoked for example.

Reply:
But the service knows what came before, so the second provider could signal that it was canceled and it should open the one before. For example, `onClose: Event<{ reopenLast: boolean }>`?

Reply:
The issue is we only want to reopen last `onClose` for certain cases. And the provider isn't aware of what those cases are.

Reply:
I'm not sure what you mean by "second provider" 🤔 . The parties of interest here are the original provider and the quick-pick / action that interrupts / hides it temporarily.

Reply:
Oh I get it. It just feels a little flimsy right now and easily breakable. Could we do something to make it more reliable? Tests that verify it interacts nicely with the second quick pick would be good

Reply:
Hmm, I think smoke tests would be needed for that?

Reply:
could be good, but I can just imagine the flakes now... 😅

Reply:
Smoke tests would be best yeah. I would think they'd be fairly reliable if all they do is verify the view is up, not validate the contents.